src/biokbase/narrative/jobs/job.py

```diff
@@ -322,21 +303,47 @@ def state(self, force_refresh=False):
         if self.was_terminal and not force_refresh:
             state = copy.deepcopy(self._acc_state)
         else:
-            try:
-                state = clients.get("execution_engine2").check_job(
-                    {"job_id": self.job_id, "exclude_fields": EXCLUDED_JOB_STATE_FIELDS}
-                )
-                state = self.update_state(state)
-
-            except Exception as e:
-                raise Exception(f"Unable to fetch info for job {self.job_id} - {e}")
+            state = self.query_ee2_state(self.job_id, False)
+            state = self.update_state(state)
 
         for field in EXCLUDED_JOB_STATE_FIELDS:
             if field in state and field != "job_input":
                 del state[field]
 
         return state
 
+    @staticmethod
+    def query_ee2_state(
+        job_id: str,
+        init: bool = True,
+    ) -> dict:
+        return clients.get("execution_engine2").check_job(
+            {
+                "job_id": job_id,
+                "exclude_fields": (
+                    JOB_INIT_EXCLUDED_JOB_STATE_FIELDS
+                    if init else
+                    EXCLUDED_JOB_STATE_FIELDS
+                )
+            }
+        )
+
+    @staticmethod
+    def query_ee2_states(
+        job_ids: List[str],
+        init: bool = True,
+    ) -> dict:
+        return clients.get("execution_engine2").check_jobs(
+            {
+                "job_ids": job_ids,
```

Comment:
if you add the argument `return_list: 0`, ee2 will return a dictionary, which saves you some trouble!